### PR TITLE
Cleanup linux headers

### PR DESCRIPTION
**Pull request by lucasdemarchi**

### CODE REVIEW

File: libraries/AP_HAL_Linux/RCInput_AioPRU.h [hunk 1]
```diff
@@ -18,16 +18,18 @@
...
-{
+namespace Linux {
+
+class RCInput_AioPRU : public RCInput {
 public:
     void init();
     void _timer_tick(void);
 
- private:
+protected:
```
Comment (guludo) [inline libraries/AP_HAL_Linux/RCInput_AioPRU.h]: Why changing the access level?

File: libraries/AP_HAL_Linux/RCInput.h [hunk 2]
```diff
@@ -31,16 +33,15 @@ class Linux::RCInput : public AP_HAL::RCInput {
...
-    
-    
- protected:
+
+
+protected:
     void _process_rc_pulse(uint16_t width_s0, uint16_t width_s1);
     void _update_periods(uint16_t *periods, uint8_t len);
 
- private:
```
Comment (guludo) [inline libraries/AP_HAL_Linux/RCInput.h]: Why changing the access level?

File: libraries/AP_HAL_Linux/RCInput_PRU.cpp [hunk 3]
```diff
@@ -17,7 +18,8 @@
 #include <unistd.h>
 
 #include "GPIO.h"
-#include "RCInput.h"
+
+#define RCIN_PRUSS_SHAREDRAM_BASE   0x4a312000
```
Comment (guludo) [inline libraries/AP_HAL_Linux/RCInput_PRU.cpp]: I understand that this helps on avoiding recompilation when RCIN_PRUSS_SHAREDRAM_BASE changes. I think this either should be a separate commit or the commit message should have a note that it has also other minor changes to avoid unnecessary recompilation.

File: libraries/AP_HAL_Linux/RCInput_RPI.h [hunk 4]
```diff
@@ -45,8 +46,8 @@ typedef struct {
 
 class Memory_table {
```
Comment (guludo) [inline libraries/AP_HAL_Linux/RCInput_RPI.h]: Is `Memory_table` supposed to be in the namespace?

Comment (guludo) [inline libraries/AP_HAL_Linux/RCInput_RPI.h]: Same for the other declarations.

Comment (guludo) [inline libraries/AP_HAL_Linux/RCInput_RPI.h]: Also, I haven't checked on the previous files.

Comment (lucasdemarchi) [inline libraries/AP_HAL_Linux/RCInput_RPI.h]: No reason for them to be outside. Or inside. So I preferred them inside.

Comment (guludo) [inline libraries/AP_HAL_Linux/RCInput_RPI.h]: Fair enough.

**Comment by guludo** [comment]
Added some minor comments. But overall looks good to me.

**Comment by lucasdemarchi** [comment]
I fixed the comments and force pushed. Let's wait for CI.
